### PR TITLE
Add UnauthenticatedTransportSessionProtocol

### PR DESCRIPTION
**Pull request by typfel**

### CODE REVIEW

File: Source/TransportSession/UnauthenticatedTransportSession.swift [hunk 1]
```diff
@@ -17,26 +17,36 @@
...
+    
+}
+
 public struct UserInfo {
-    let identifier: UUID
-    let cookieData: Data
+    public let identifier: UUID
+    public let cookieData: Data
+    
+    public init(identifier: UUID, cookieData: Data) {
```
Comment (daehn) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: The memberwise `init` should be autogenerated for structs if I'm not mistaken.

Comment (typfel) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: It's scoped `internal` though and I need to use it in the mock transport.

File: Source/TransportSession/UnauthenticatedTransportSession.swift [hunk 2]
```diff
@@ -49,7 +59,7 @@ final public class UnauthenticatedTransportSession: NSObject {
 
     public init(
         baseURL: URL,
-        delegate: UnauthenticatedTransportSessionDelegate,
+        delegate: UnauthenticatedTransportSessionDelegate? = nil,
```
Comment (daehn) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: I am not sure it makes sense for a caller to pass `nil` here. Why create an `UnauthenticatedTransportSession` when not wanting to get a callback when there is a `UserInfo`?

File: Source/TransportSession/UnauthenticatedTransportSession.swift (hunk 1 above)
Comment (daehn) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: I see, interesting that the `init` is `internal` by default even though the struct itself is `public`.

File: Source/TransportSession/UnauthenticatedTransportSession.swift (hunk 2 above)
Comment (typfel) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: Hmm I think I changed due to wanting to inject the `UnauthenticatedTransportSession` into the  `UnauthenticatedSession`

Comment (daehn) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: Ah ok, maybe we can just remove it from the `init` then.

Comment (typfel) [inline Source/TransportSession/UnauthenticatedTransportSession.swift]: Yes, maybe. But I think it makes sense to make interface changes while having the whole setup wired together, it's still in a bit of flux atm.

**Comment by typfel** [comment]
Hmm don't understand why the server trust test started to fail on this branch

**Comment by daehn** [comment]
Is that one hitting the network?

**Comment by daehn** [comment]
Happens in the base branch as well, let's check and fix it there.

**Comment by typfel** [comment]
Pull in develop. `ZMServerTrust.m` has been modified there.

**Pull request merged**